lib/ansible/galaxy/collection/gpg.py

```diff
@@ -37,7 +37,7 @@ def get_signature_from_source(source, display=None):  # type: (str, t.Optional[D
             f"Failed to get signature for collection verification from '{source}': {e}"
         ) from e
 
-    return signature
+    return signature.decode('utf-8')
```

Comment:
Perhaps, add type annotations to this function signature before making runtime changes? It's not being type-checked unless you do so. Let MyPy show where your change may possibly cause problems.

Reply:
It has type annotations, which is how I knew it caused an issue. It was that `open_url` didn't so it had no inference about what was being returned.